src/wallets/solana/index.ts

```
@@ -167,16 +172,34 @@ export class SolanaWalletToolbox extends WalletToolbox {
     }, this.options.tokenPollConcurrency)
   }
 
-  async transferNativeBalance(sourceAddress: string, targetAddress: string, amount: number): Promise<TransferRecepit> {
+  protected async transferNativeBalance(sourceAddress: string, targetAddress: string, amount: number): Promise<TransferRecepit> {
     // TODO: implement
     throw new Error('SolanaWalletToolbox.transferNativeBalance not implemented.');
   }
 
-  getAddressFromPrivateKey(privateKey: string): string {
+  protected async getRawWallet (privateKey: string) {
+    let secretKey;
+      try {
+        secretKey = decode(privateKey);
+      } catch (e) {
+        secretKey = new Uint8Array(JSON.parse(privateKey));
+      }
+    return {
+      conn: this.connection,
+      payer: Keypair.fromSecretKey(secretKey),
+    } as SolanaWallet;
+  }
+
+  public async getGasPrice () {
```

should we throw instead of returning 0?

---

nice catch! We should actually throw error here, until we develop those methods. I will update the code.